src/qt/askpassphrasedialog.cpp

```diff
@@ -119,7 +120,12 @@ void AskPassphraseDialog::accept()
         {
             if(newpass1 == newpass2)
             {
-                if(model->setWalletEncrypted(true, newpass1))
+                bool success = (model && model->setWalletEncrypted(true, newpass1)) || m_passphrase_out;
+                if (m_passphrase_out)
+                {
+                    m_passphrase_out->assign(newpass1);
```

Comment:
I suggest you add a `QMessageBox::warning` here (the message in the existing success warning about "Any previous backups" is not relevant for new wallets) and then return.

Reply:
Done